### DISCUSSION TITLE
Share variables between sessions

### DISCUSSION THREAD

**Discussion by rarealphacat** [scandum/tintin#81] [General]
Is there a way to share variables between sessions simultaneously? I share variables between sessions with #write now, hoping to find a faster method. #session might help but it wouldn't work if I open 2 tintin in different terminals

**Comment by scandum** [comment]
There are a few options.
```
#all #variable test 1
```
Would set the variable test in all sessions except for the startup session.
```
#gts #variable test 1
#showme @gts{$test}
```
This would set the variable test in the startup session, and with `@gts{$test}` you would pull the value from the startup session. `gts` is the name of the startup session.

**Comment by rarealphacat** [comment]
oh thank you, #all works :D Do you know any magical ways to share variables between 2 tintin++ windows (terminals)?

**Reply by scandum** [reply]
There's no real easy way to do so. #port is probably the best way.

There's the satellite script which is a little convoluted that might give some insights.

https://tintin.mudhalla.net/scripts/satellite.php

#port is pretty barebone, it just allows the creation of a socket connection between two tintin++ instances, you'd have to do some scripting, likely using an #action, to exchange and set variables.

**Comment by rarealphacat** [comment]
Thanks alot, you gave some some new ideas.
